### ISSUE TITLE
ingester name should match indexer name layout

### ISSUE THREAD

**Issue by fsgeek** [ubc-systopia/Indaleko#10]
```text
indaleko-platform=Windows-indexer=local_fs-machine=2e169bb7_0024_4dc1_93dc_18b7d2d28190-storage=3397d97b_2ca5_11ed_b2fc_b40ede9a5a3c-timestamp=2024-01-18T19-55-18.491749+00-00.jsonl
indaleko-Windows-ingester=local-fs-ingester-machine=2e169bb7-0024-4dc1-93dc-18b7d2d28190-storage=3397d97b-2ca5-11ed-b2fc-b40ede9a5a3c-2024-01-18T19-55-18.491749+00-00..jsonl
```

Note there is a "platform" label in the first but not the second, there is no label for the generator in either name, which is fine so long as the pattern is the same. The idea behind uniform naming is to make it easy to extract a key/value list of these fields.

Fix this to be more consistent.

**Comment by fsgeek** [comment]
So, the "fix" for this was to build a standard mechanism for generating these names. It's still flexible, but it has structure so values can be embedded within it.  The other places where names are generated can then be converted to use this uniform generator and extractor.

**Comment by fsgeek** [comment]
Changes for this are in commit 4a0702c.  They touch multiple files because in addition to unifying the creation and extraction I had to modify the way various fields were being handle.  Thus, UUIDs are now hex representations rather than the hyphen representation that is typical.  Default values are now 0 instead of "unknown".  While I was at it, I changed name generation to allow overriding the suffix, which means log file names now also sync up with the other file names without using the "replace" hack.
